packages/data/addon/serializers/metadata/bard.ts

```diff
@@ -0,0 +1,237 @@
+          timegrain.dimensions.forEach(dimension => {
+            const { datatype: valueType } = dimension;
+            const accDimensionList = valueType === 'date' ? currentTimeDimensions : currentDimensions;
+            const accTableDimensionList = valueType === 'date' ? tableTimeDimensionIds : tableDimensionIds;
+
+            const newDim = this._constructDimension(dimension, source);
+            const newDimCardinality = newDim.cardinality || CARDINALITY_SIZES[2];
+            if (CARDINALITY_SIZES.indexOf(newDimCardinality) > CARDINALITY_SIZES.indexOf(acc.tableCardinality)) {
+              acc.tableCardinality = newDimCardinality;
+            }
+            accDimensionList[newDim.id] = newDim; // Add dim to all dimensions list
+            accTableDimensionList.add(newDim.id); // Add dim id to table's dimensionIds/timeDimensionIds list
+          });
+
+          // Construct each metric and metric function + function arguments if necessary
+          timegrain.metrics.forEach((metric: RawMetricPayload) => {
+            const convertedToMetricFunction = this._getMetricFunctionFromParameters(metric, source);
+            if (convertedToMetricFunction) {
+              metric.metricFunctionId = convertedToMetricFunction.id;
+            }
+
+            const newMetric = this._constructMetric(metric, source);
+            currentMetrics[newMetric.id] = newMetric; // Add metric to all metrics list
+            tableMetricIds.add(newMetric.id); // Add metric id to table's metricIds list
+
+            if (convertedToMetricFunction) {
+              convertedToMetricFunctions[convertedToMetricFunction.id] = convertedToMetricFunction;
+            }
+          });
+
+          return acc;
+        },
+        {
+          metrics,
+          dimensions,
+          timeDimensions,
+          tableMetricIds: new Set<string>(),
+          tableDimensionIds: new Set<string>(),
+          tableTimeDimensionIds: new Set<string>(),
+          tableCardinality: CARDINALITY_SIZES[0] as typeof CARDINALITY_SIZES[number]
+        }
+      );
+
+      return {
+        id: table.name,
+        name: table.longName,
+        description: table.description,
+        category: table.category,
+        cardinality: allTableColumns.tableCardinality,
+        timeGrainIds: table.timeGrains.map(grain => grain.name),
+        source,
+        metricIds: [...allTableColumns.tableMetricIds],
+        dimensionIds: [...allTableColumns.tableDimensionIds],
+        timeDimensionIds: [...allTableColumns.tableTimeDimensionIds]
+      };
+    });
+
+    const metricFunctions = rawMetricFunctions ? normalizeMetricFunctions(rawMetricFunctions, source) : [];
+
+    return {
+      tables,
+      dimensions: Object.values(dimensions),
+      metrics: Object.values(metrics),
+      timeDimensions: Object.values(timeDimensions),
+      metricFunctions: [...metricFunctions, ...Object.values(convertedToMetricFunctions)]
```

Comment:
Yeah, this will only happen if there are metrics with metric IDs and metrics with **only** `parameters` in a single data source. 
https://github.com/yahoo/navi/pull/931/files#diff-60c8edbf89613347ba5e51e792ff0808R180